rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -88,30 +91,23 @@ impl Default for TunnelManager {
     }
 }
 
-pub struct GetWgInterface(pub IpAddr);
-impl Message for GetWgInterface {
-    type Result = Result<TunnelData, Error>;
+pub struct IdentityCallback(pub LocalIdentity, pub Peer);
+impl Message for IdentityCallback {
+    type Result = Option<Tunnel>;
 }
 
-impl Handler<GetWgInterface> for TunnelManager {
-    type Result = Result<TunnelData, Error>;
-
-    fn handle(&mut self, msg: GetWgInterface, _: &mut Context<Self>) -> Self::Result {
-        Ok(self.get_if(msg.0))
-    }
-}
+impl Handler<IdentityCallback> for TunnelManager {
+    type Result = Option<Tunnel>;
 
-pub struct Listen(pub String);
-impl Message for Listen {
-    type Result = ();
-}
-
-impl Handler<Listen> for TunnelManager {
-    type Result = ();
-
-    fn handle(&mut self, listen: Listen, _: &mut Context<Self>) -> Self::Result {
-        self.listen_interfaces.insert(listen.0);
-        SETTING.get_network_mut().peer_interfaces = self.listen_interfaces.clone();
+    fn handle(&mut self, msg: IdentityCallback, _: &mut Context<Self>) -> Self::Result {
+        let local_id = msg.0;
+        let peer = msg.1;
+        let res = self.open_tunnel(local_id, peer);
+        if res.is_err() {
```

Comment:
I think the pattern below can be applied to this section and many other sections of code, so you don't do unwraps, even if you know they will work (makes for easier grepping of unwraps to find potential failure points)
```
match res {
    Ok(res) => {
        Some(res)
    }
    Err(e) => {
        warn!("Open Tunnel failed with {:?}", e);
    }
}
```